schema/function_test.go

```diff
@@ -0,0 +1,517 @@
+package schema_test
+
+import (
+	"fmt"
+	"go.arcalot.io/assert"
+	"go.flow.arcalot.io/pluginsdk/schema"
+	"reflect"
+	"testing"
+)
+
+// Simple, no input or output.
+func TestCallableFunctionSchema_Simple(t *testing.T) {
+	called := false
+	simpleFunc, err := schema.NewCallableFunction(
+		"test",
+		make([]schema.Type, 0),
+		nil,
+		nil,
+		func() {
+			called = true
+		},
+	)
+	assert.NoError(t, err)
+	assert.Equals(t, called, false)
+	result, err := simpleFunc.Call([]any{})
+	assert.NoError(t, err)
+	assert.Nil(t, result)
+	assert.Equals(t, called, true)
+}
+
+// Simple, with one input.
+func TestCallableFunctionSchema_Simple1Param(t *testing.T) {
+	passedInVal := ""
+	simpleFunc, err := schema.NewCallableFunction(
+		"test",
+		[]schema.Type{schema.NewStringSchema(nil, nil, nil)},
+		nil,
+		nil,
+		func(test string) {
+			passedInVal = test
+		},
+	)
+	assert.NoError(t, err)
+	result, err := simpleFunc.Call([]any{"a"})
```

Comment:
I don't think there's a need to overcomplicate the unit tests. There are no variables that would make it so that mechanisms tested in the prior test differ here.

Reply:
I don't think there is _ever_ a need for **over**-complication!  🤣 

That aside....

> There are no variables that would make it so that mechanisms tested in the prior test differ here.

_You_ know that, and you know that it is true _at the moment_; however, the _test_ (e.g., "as a specification") doesn't "know" that, and it probably shouldn't presume it.  And, we hope that the test code will be longer-lived than the source code that it tests, such that known truths today many change in the future.  So, we really should be testing the behavior that we intend the code to follow, independent of any knowledge that we currently have of its actual implementation.

Also, test code has been known to evolve as well, and tests are usually evaluated on a per-test basis rather than as a part of a whole -- that is, it would be unfortunate if the previous test were removed or modified such that we lost the only place where that behavior was validated.  Another way of putting this is, each test should test "one thing", and it should be clear what its purpose is.  Thus, if the not-called-early behavior is a thing unto itself, it should have a separate test; otherwise, it should be checked by each of the tests to which it is pertinent.

Reply:
In the case that uses passedInVal, the function doesn't have a way to output its input, since it isolates just input in this test specifically. Once we check for that, that test is enough. I do not think it's worth it for the potential evolution.